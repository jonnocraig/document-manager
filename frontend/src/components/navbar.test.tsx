import * as React from 'react';
import { shallow } from 'enzyme';
import Nav from './navbar';

function setup(nextProps = {}) {
  const props = {
    ...nextProps
  };
  const wrapper = shallow(<Nav {...props} />);

  return {
    props,
    wrapper
  };
}

describe('<Nav />', () => {
  it('should match the snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
