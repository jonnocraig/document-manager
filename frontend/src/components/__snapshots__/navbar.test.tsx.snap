// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <NavbarBrand
        href="/"
        tag="a"
      >
        <img
          className="logo"
        />
        <h1
          className="logoHeading"
        >
          Document Manager
        </h1>
      </NavbarBrand>,
      "color": "dark",
      "dark": true,
      "expand": "md",
      "tag": "nav",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <img
            className="logo"
          />,
          <h1
            className="logoHeading"
          >
            Document Manager
          </h1>,
        ],
        "href": "/",
        "tag": "a",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "logo",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Document Manager",
            "className": "logoHeading",
          },
          "ref": null,
          "rendered": "Document Manager",
          "type": "h1",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <NavbarBrand
          href="/"
          tag="a"
        >
          <img
            className="logo"
          />
          <h1
            className="logoHeading"
          >
            Document Manager
          </h1>
        </NavbarBrand>,
        "color": "dark",
        "dark": true,
        "expand": "md",
        "tag": "nav",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <img
              className="logo"
            />,
            <h1
              className="logoHeading"
            >
              Document Manager
            </h1>,
          ],
          "href": "/",
          "tag": "a",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "logo",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Document Manager",
              "className": "logoHeading",
            },
            "ref": null,
            "rendered": "Document Manager",
            "type": "h1",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
